src/main/scala/io/confluent/examples/streams/WordCountScalaExample.scala

```diff
@@ -0,0 +1,134 @@
+  * Once packaged you can then run:
+  *
+  * {{{
+  * $ java -cp target/kafka-streams-examples-5.0.0-SNAPSHOT-standalone.jar io.confluent.examples.streams.WordCountLambdaExample
+  * }}}
+  *
+  * 4) Write some input data to the source topic "streams-plaintext-input" (e.g. via `kafka-console-producer`).
+  * The already running example application (step 3) will automatically process this input data and write the
+  * results to the output topic "streams-wordcount-output".
+  *
+  * {{{
+  * # Start the console producer. You can then enter input data by writing some line of text, followed by ENTER:
+  * #
+  * #   hello kafka streams<ENTER>
+  * #   all streams lead to kafka<ENTER>
+  * #   join kafka summit<ENTER>
+  * #
+  * # Every line you enter will become the value of a single Kafka message.
+  * $ bin/kafka-console-producer --broker-list localhost:9092 --topic streams-plaintext-input
+  * }}}
+  *
+  * 5) Inspect the resulting data in the output topic, e.g. via `kafka-console-consumer`.
+  * {{{
+  * $ bin/kafka-console-consumer --topic streams-wordcount-output --from-beginning \
+  *                              --new-consumer --bootstrap-server localhost:9092 \
+  *                              --property print.key=true \
+  *                              --property value.deserializer=org.apache.kafka.common.serialization.LongDeserializer
+  * }}}
+  * You should see output data similar to below. Please note that the exact output
+  * sequence will depend on how fast you type the above sentences. If you type them
+  * slowly, you are likely to get each count update, e.g., kafka 1, kafka 2, kafka 3.
+  * If you type them quickly, you are likely to get fewer count updates, e.g., just kafka 3.
+  * This is because the commit interval is set to 10 seconds. Anything typed within
+  * that interval will be compacted in memory.
+  *
+  * {{{
+  * hello    1
+  * kafka    1
+  * streams  1
+  * all      1
+  * streams  2
+  * lead     1
+  * to       1
+  * join     1
+  * kafka    3
+  * summit   1
+  * }}}
+  *
+  * 6) Once you're done with your experiments, you can stop this example via `Ctrl-C`. If needed,
+  * also stop the Kafka broker (`Ctrl-C`), and only then stop the ZooKeeper instance (`Ctrl-C`).
+  **/
+object WordCountScalaExample extends App {
+
+  import org.apache.kafka.streams.scala.DefaultSerdes._
+  import org.apache.kafka.streams.scala.ImplicitConversions._
+
+  val config: Properties = {
+    val p = new Properties()
+    p.put(StreamsConfig.APPLICATION_ID_CONFIG, "wordcount-scala-application")
+    val bootstrapServers = if (args.length > 0) args(0) else "localhost:9092"
+    p.put(StreamsConfig.BOOTSTRAP_SERVERS_CONFIG, bootstrapServers)
+    p
+  }
+
+  val builder: StreamsBuilder = new StreamsBuilder()
+  val textLines: KStream[String, String] = builder.stream[String, String]("streams-plaintext-input")
+  val wordCounts: KTable[String, Long] = textLines
```

Comment:
do we need to specify the type?

Reply:
See above.  No, we don't.  But it helps to make it clear that the result type of the operation is a KTable -- most beginners would otherwise probably assume it's still a KStream. That's why I prefer to keep explicit type information in the examples.